modules/shortcodes/polldaddy.php

```diff
@@ -13,12 +14,91 @@ class PolldaddyShortcode {
+			add_filter( 'pre_kses', array( $this, 'polldaddy_embed_to_shortcode' ) );
+		}
 		add_action( 'wp_enqueue_scripts', array( $this, 'check_infinite' ) );
 		add_action( 'infinite_scroll_render', array( $this, 'polldaddy_shortcode_infinite' ), 11 );
 	}
 
+	private function get_async_code( array $settings, $survey_link ) {
+		$embed_src     = 'http://i0.poll.fm/survey.js';
+		$embed_src_ssl = 'https://polldaddy.com/survey.js';
+
+		$include = <<<CONTAINER
+( function( d, c, j ) {
+  if ( !d.getElementById( j ) ) {
+    var pd = d.createElement( c ), s;
+    pd.id = j;
+    pd.src = ( 'https:' == d.location.protocol ) ? '{$embed_src_ssl}' : '{$embed_src}';
+    s = d.getElementsByTagName( c )[0];
+    s.parentNode.insertBefore( pd, s );
+  }
+}( document, 'script', 'pd-embed' ) );
+CONTAINER;
+
+		// Compress it a bit
+		$include = $this->compress_it( $include );
+
+		$placeholder = '<div class="pd-embed" data-settings="'.esc_attr( json_encode( $settings ) ).'"></div>';
+		if ( $type === 'button' )
+			$placeholder = '<a class="pd-embed" href="'.esc_attr( $survey_link ).'" data-settings="'.esc_attr( json_encode( $settings ) ).'">'.esc_html( $settings['title'] ).'</a>';
+
+		$js_include = $placeholder."\n";
+		$js_include .= '<script type="text/javascript"><!--//--><![CDATA[//><!--'."\n";
+		$js_include .= $include."\n";
+		$js_include .= "//--><!]]></script>\n";
+
+		if ( $type !== 'button' )
+			$js_include .= '<noscript>'.$survey_link."</noscript>\n";
+
+		return $js_include;
+	}
+
+	private function compress_it( $js ) {
+		$js = str_replace( array( "\n", "\t", "\r" ), '', $js );
+		$js = preg_replace( '/\s*([,:\?\{;\-=\(\)])\s*/', '$1', $js );
+		return $js;
+	}
+	
+	/*
+	 * Polldaddy Poll Embed script - transforms code that looks like that:
+	 * <script type="text/javascript" charset="utf-8" src="http://static.polldaddy.com/p/123456.js"></script>
+	<noscript><a href="http://polldaddy.com/poll/123456/">What is your favourite color?</a></noscript>
+	 * into the [polldaddy poll=...] shortcode format
+	 */
+	function polldaddy_embed_to_shortcode( $content ) {
+				
+		if ( false === strpos( $content, 'polldaddy.com/p/' ) )
+			return $content;
+
+		$regexes = array();
+
+		$regexes[] = '#<script[^>]+?src="https?://(secure|static)\.polldaddy\.com/p/([0-9]+)\.js"[^>]*+>\s*?</script>\r?\n?(<noscript>.*?</noscript>)?#i';
+
+		$regexes[] = '#&lt;script(?:[^&]|&(?!gt;))+?src="https?://(secure|static)\.polldaddy\.com/p/([0-9]+)\.js"(?:[^&]|&(?!gt;))*+&gt;\s*?&lt;/script&gt;\r?\n?(&lt;noscript&gt;.*?&lt;/noscript&gt;)?#i';
+	
+		foreach ( $regexes as $regex ) {
+			if ( ! preg_match_all( $regex, $content, $matches, PREG_SET_ORDER ) )
+		 		continue;
+	
+			foreach ( $matches as $match ) {
+				if (!isset( $match[2] ) )
+					continue;
+					
+				$id = (int) $match[2];
+				
+				if ( $id > 0 ) {
+					$content = str_replace( $match[0], "[polldaddy poll=$id]", $content );
+					do_action( 'jetpack_embed_to_shortcode', 'polldaddy', $url );
```

Comment:
Need to add preceding
`/** This action is documented in modules/shortcodes/youtube.php */`